CS3219-AY2425S1/cs3219-ay2425s1-project-g29
# Added Attempt History Table for each User and Account Deletion Functionality

## Changes Made:
- Implement Account Delete Functionality (both password & Google accounts)
- Implement User Attempt History Table in each User's Profile Page
  - Connect User Service to Fire Store
  - Fire Store now stores attempt history
  - Matching Service updates User Service when matches occur
- Refactor verify_token functions in User Service
- Remove unnecessary refreshUser call from Navbar
- Fixed bug where the login page doesn't load on the first load (because of Navbar trying to access the profile picture while the user is logged out)

## More Info
Firestore now stores attempt history for each user. When a match occurs, matching service tells user service who the matched users are, the time and the question.

User Service stores this match in Firestore in this manner.
attempt_history -> (user_id) -> attempts -> (session_id)
With this format:
```
matched_user: "user_id" (string)
question_id: "question_id" (string)
timestamp: "1729500216.228071": (string)
```
So when a match occurs, User Service creates two new entries for both of the matched users involved.

## How to test:
Login to two instances of PeerPrep, and start a match between both. When a match is found, the attempt history in each of the user's profile page should be updated with the latest match.

Fixes #7
